src/EFCore.PG.NTS/Storage/Internal/NpgsqlNetTopologySuiteTypeMappingSourcePlugin.cs

```diff
@@ -21,17 +36,96 @@ public NpgsqlNetTopologySuiteTypeMappingSourcePlugin([NotNull] INpgsqlNetTopolog
         public virtual RelationalTypeMapping FindMapping(in RelationalTypeMappingInfo mappingInfo)
         {
             var clrType = mappingInfo.ClrType;
-            var storeTypeName = mappingInfo.StoreTypeName;
+            var storeType = mappingInfo.StoreTypeName;
 
             // TODO: Array
-            return clrType != null && typeof(IGeometry).IsAssignableFrom(clrType) ||
-                   storeTypeName != null && (
-                        storeTypeName.Equals("geometry", StringComparison.OrdinalIgnoreCase) ||
-                        storeTypeName.Equals("geography", StringComparison.OrdinalIgnoreCase))
-                ? (RelationalTypeMapping)Activator.CreateInstance(
-                    typeof(NpgsqlGeometryTypeMapping<>).MakeGenericType(clrType ?? typeof(IGeometry)),
-                    storeTypeName ?? (_options.IsGeographyDefault ? "geography" : "geometry"))
-                : null;
+            // TODO: SRID (https://github.com/aspnet/EntityFrameworkCore/issues/14000)
+
+            if (clrType != null)
+            {
+                if (typeof(IGeometry).IsAssignableFrom(clrType) &&
+                    (storeType != null || TryGetStoreType(clrType, _options.IsGeographyDefault, null, out storeType)))
+                {
+                    return (RelationalTypeMapping)Activator.CreateInstance(
+                        typeof(NpgsqlGeometryTypeMapping<>).MakeGenericType(clrType),
+                        storeType);
+                }
+            }
+            else if (storeType != null)
+            {
+                var x = ParseGeometryStoreType(storeType);
+                if (!x.HasValue)
+                    return null;
+
+                if (!_storeTypeMappings.TryGetValue(x.Value.SpatialType ?? "geometry", out clrType))
+                    return null;
+
+                return (RelationalTypeMapping)Activator.CreateInstance(
+                    typeof(NpgsqlGeometryTypeMapping<>).MakeGenericType(clrType),
+                    storeType);
+            }
+
+            return null;
+        }
+
+        static readonly Regex _storeTypeRegex = new Regex(@"^(GEOMETRY|GEOGRAPHY)\s*(\((\w+)(,(\d+))?\))?$", RegexOptions.IgnoreCase);
```

Comment:
Ehhh....I'm sure this is fine...but it feels out of place. 

For comparison, [`RelationalTypeMappingInfo.ParseStoreTypeName(...)`](https://github.com/aspnet/EntityFrameworkCore/blob/5262ff5b16fffd0241975fb0b90b01cb71966fd2/src/EFCore.Relational/Storage/RelationalTypeMappingInfo.cs#L190-L244) avoids regex and parses directly.

Reply:
Yeah, I considered doing it by hand, but this case seemed a bit more complex than usual and I guess I'm a bit too partial to regex :) I don't see any concrete reason to avoid regex though - in any case it's an internal implementation detail we can always change if it becomes troublesome...

Reply:
Makes sense, just wanted to flag it for discussion.